js/ui/status/keyboard.js

```diff
@@ -22,6 +22,15 @@ const Util = imports.misc.util;
 const INPUT_SOURCE_TYPE_XKB = 'xkb';
 const INPUT_SOURCE_TYPE_IBUS = 'ibus';
 
+const InputSourceManagerIface = '<node> \
+<interface name="org.gnome.Shell.InputSourceManager"> \
+<method name="SetPasswordModeEnabled"> \
+ <arg name="enabled" type="b" direction="in"/> \
+</method> \
+<property name="PasswordModeEnabled" type="b" access="read" /> \
```

Comment:
This could be a `readwrite` property instead, which would save you from writing a separate method just to set it.

Reply:
I had this read-write before. I tried to use g_dbus_proxy_set_cached_property like in other places for it, but that was not received by the shell. So I used code I found for Enabling Metrics 'SetEnabled' in the control center https://github.com/endlessm/gnome-control-center/blob/master/panels/privacy/cc-privacy-panel.c#L907.

Reply:
`g_dbus_proxy_set_cached_property()` won't help, as it will only set the value of the local cached property of the proxy, not send it over the wire.
You will need to call the `Set` method on the `org.freedesktop.DBus.Properties` interface from the client side. (Or alternatively use `gdbus-codegen` to generate a class-specific proxy that actually has a property setter that will call that for you, but that's a lot of yak shaving.)